### PR TITLE
Polycrystal Finance && Timewarp Protocol && Fabric && OpenOcean && Stormswap && pTokens TVL Adapters

### PR DESCRIPTION
**Pull request by Mohammed18062020**
##### Twitter Link:
Polycrystal Finance -> https://twitter.com/PolyCrystal_
Timewarp protocol -> https://twitter.com/ChronotechNews
Fabric -> https://twitter.com/official_fabric?lang=en
OpenOcean -> https://twitter.com/OpenOceanGlobal
Stormswap -> https://twitter.com/StormSwap_Fi
pTokens -> https://twitter.com/pNetworkDefi

##### Website Link:
Polycrystal Finance -> https://www.polycrystal.finance/
Timewarp protocol -> https://timewarp.finance/
Fabric -> https://stake.fsynth.io/#/
OpenOcean -> https://openocean.finance/
Stormswap -> https://stormswap.finance/
pTokens -> https://ptokens.io/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
Polycrystal Finance -> 
image

Timewarp protocol -> 
image
Fabric -> 
image

OpenOcean -> 
image

Stormswap -> 
image

pTokens -> 
image

##### Current TVL:
Polycrystal Finance -> $18,709,934.43
Timewarp protocol -> $23,200,330
Fabric -> $1,575,218.991
OpenOcean -> $7.52M
Stormswap -> $37,077,765.65
pTokens -> $ 189,908,479.60

##### Chain:
Polycrystal Finance -> Polygon
Timewarp protocol -> Ethereum and Binance
Fabric -> Solana
OpenOcean -> BSC and Thereum
Stormswap -> Avalanche
pTokens -> Ethereum, Bsc, Eos, Telos, Ultra, Polygon and Xdai

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
Polycrystal Finance -> "id": "polycrystal"
Timewarp protocol -> "id": "chronobank"
Fabric -> "id": "fabric"
OpenOcean -> "id": "openocean"
Stormswap -> 
pTokens -> "id": "pnetwork"

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
Polycrystal Finance -> 
Timewarp protocol -> "id":1556
Fabric -> "id":11462
OpenOcean -> "id":1261
Stormswap -> 
pTokens -> "id":5794

##### Token address and ticker if any:
Polycrystal Token(CRYSTAL) -> 0x76bF0C28e604CC3fE9967c83b3C3F31c213cfE64
Timewarp / ChronoTech Token : 
     ETH (TIME) -> 0x485d17a6f1b8780392d53d64751824253011a260
     BSC (TIME) -> 0x3b198e26E473b8faB2085b37978e36c9DE5D7f68
Fabric Token (FAB) -> EdAhkbj5nF9sRM7XN7ewuW8C9XEUMs8P7cnoQ57SYE96
OpenOcean Token (OOE) -> 0x7778360f035c589fce2f4ea5786cbd8b36e5396b
Stormswap Token (STORM) -> 0x6AFD5A1ea4b793CC1526d6Dc7e99A608b356eF7b
pTokens Token (PNT):
         ETH -> 0x89ab32156e46f46d02ade3fecbe5fc4243b9aaed
         BSC -> 0xdaacb0ab6fb34d24e8a67bfa14bf4d95d4c7af92
 

##### Category (Yield/Dexes/Lending/Minting/Assets/Insurance/Options/Indexes/Staking) *Please choose only one:
Polycrystal Finance -> Yield
Timewarp protocol -> Staking
Fabric -> Yield
OpenOcean -> Staking
Stormswap -> Staking
pTokens -> Assets

### CODE REVIEW

**Comment by Mohammed18062020** [comment]
TimeWarp protocolo is a product of ChronoTech Ecosystem to do staking on Ethereum and Binance.

**Comment by Mohammed18062020** [comment]
Four more tvl adapters on different chains.

File: projects/fabric/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    .map((sup) => sup.liquidity)
+    .forEach(function (sup) {
+      totalLiquidityUSD += sup;
+    });
+
+  return toUSDTBalances(totalLiquidityUSD);
+}
+
+module.exports = {
+  solona: {
```
Comment (0xngmi) [inline projects/fabric/index.js:20]: the correct spelling is `solana`

File: projects/pTokens/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+const sdk = require("@defillama/sdk");
+const utils = require("../helper/utils");
+const { toUSDTBalances } = require("../helper/balances");
+
+const API_URL = "https://chart.ptokens.io/index.php?a=assets";
+
+async function calcTvl(CHAIN) {
+  let totalLiquidityUSD = 0;
+
+  (await utils.fetchURL(API_URL)).data
```
Comment (0xngmi) [inline projects/pTokens/index.js:10]: would be great if this worked with historical data points too

File: projects/fabric/index.js (hunk 1 above)
Comment (Mohammed18062020) [reply projects/fabric/index.js:20]: That's right.

File: projects/pTokens/index.js (hunk 2 above)
Comment (Mohammed18062020) [reply projects/pTokens/index.js:10]: Yes, I saw the one already is done has the historical data. Then we can left that one and don't do the overwrite, I think!

**Pull request merged**